pandas/tests/generic/test_generic.py

```diff
@@ -339,6 +339,11 @@ def test_copy_and_deepcopy(self, shape, func):
         assert obj_copy is not obj
         self._compare(obj_copy, obj)
 
+    def test_where(self, cond):
```

Comment:
you don't need `cond` as an argument here - you can try running this test with
```
pytest pandas/tests/generic/test_generic.py
```

Reply:
I run into a lot of problems when I tried running pytest. 

1. Using terminal I get all green test passed:
'
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % pytest pandas/tests/generic/test_generic.py
=========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/jiezheng/pandas-jie, configfile: pyproject.toml
plugins: hypothesis-6.13.0, cov-2.12.0, xdist-2.2.1, asyncio-0.14.0, instafail-0.4.1, forked-1.3.0
collected 22 items                                                                                                                        

pandas/tests/generic/test_generic.py ......................

-------------------------------------- generated xml file: /Users/jiezheng/pandas-jie/test-data.xml ---------------------------------------
========================================================== slowest 30 durations ===========================================================
0.07s call     pandas/tests/generic/test_generic.py::TestNDFrame::test_depr_take_kwarg_is_copy[DataFrame-True]
0.01s call     pandas/tests/generic/test_generic.py::TestNDFrame::test_squeeze

(28 durations < 0.005s hidden.  Use -vv to show these durations.)
=========================================================== 22 passed in 0.17s ============================================================
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % 
' 
2. running on terminal just on my function, then I get the not found error. 

' (pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % pytest pandas/tests/generic/test_generic.py::test_where
=========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/jiezheng/pandas-jie, configfile: pyproject.toml
plugins: hypothesis-6.13.0, cov-2.12.0, xdist-2.2.1, asyncio-0.14.0, instafail-0.4.1, forked-1.3.0
collected 0 items                                                                                                                         

-------------------------------------- generated xml file: /Users/jiezheng/pandas-jie/test-data.xml ---------------------------------------
========================================================== no tests ran in 0.01s ==========================================================
ERROR: not found: /Users/jiezheng/pandas-jie/pandas/tests/generic/test_generic.py::test_where
(no name '/Users/jiezheng/pandas-jie/pandas/tests/generic/test_generic.py::test_where' in any of [<Module test_generic.py>])

(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % 

3. when using vscode, I get the following error message : 

`
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % pytest pandas/tests/generic/test_generic.py
ImportError while loading conftest '/Users/jiezheng/pandas-jie/pandas/conftest.py'.
pandas/__init__.py:22: in <module>
    from pandas.compat import (
pandas/compat/__init__.py:15: in <module>
    from pandas.compat.numpy import (
pandas/compat/numpy/__init__.py:7: in <module>
    from pandas.util.version import Version
pandas/util/__init__.py:1: in <module>
    from pandas.util._decorators import (  # noqa
pandas/util/_decorators.py:14: in <module>
    from pandas._libs.properties import cache_readonly  # noqa
pandas/_libs/__init__.py:13: in <module>
    from pandas._libs.interval import Interval
E   ModuleNotFoundError: No module named 'pandas._libs.interval'
`
Any of these are normal? I will take the 'cond' out by the way.

Reply:
Nice! Should be good then. For 2) you probably need to specify the class name as well, or you could try `pytest pandas/tests/generic/test_generic.py -k test_where` . For 3) I'm surprised that it worked in the terminal but not in vscode, but if you see an error like that it typically means you need to build the c extensions again, `python setup.py build_ext -i -j 4`

Reply:
for 2) this is what I get now: 
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % pytest pandas/tests/generic/test_generic.py -k test_where
============================= test session starts ==============================
platform darwin -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/jiezheng/pandas-jie, configfile: pyproject.toml
plugins: hypothesis-6.13.0, cov-2.12.0, xdist-2.2.1, asyncio-0.14.0, instafail-0.4.1, forked-1.3.0
collected 22 items / 22 deselected                                             

--------- generated xml file: /Users/jiezheng/pandas-jie/test-data.xml ---------
============================ 22 deselected in 0.03s ============================
I thought I should get something like 1 test passed?  Not sure how to interpret the result here? 

for 3) I tried in vscode to build the c extension using both base and panda-env 

> 

(base) jiezheng@Jies-MacBook-Air pandas-jie % python setup.py build_ext -i -j 4
  File "setup.py", line 249
    f"{extension}-source file '{sourcefile}' not found.\n"
                                                         ^
SyntaxError: invalid syntax
(base) jiezheng@Jies-MacBook-Air pandas-jie % conda activate pandas-dev                              
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % python setup.py build_ext -i -j 4
  File "setup.py", line 249
    f"{extension}-source file '{sourcefile}' not found.\n"
                                                         ^
SyntaxError: invalid syntax

> 


`
and the line 249 refers to this run function. 

> 

    def run(self):
        if "cython" in cmdclass:
            self.run_command("cython")
        else:
            # If we are not running cython then
            # compile the extensions correctly
            pyx_files = [(self._pyxfiles, "c"), (self._cpp_pyxfiles, "cpp")]

            for pyxfiles, extension in pyx_files:
                for pyxfile in pyxfiles:
                    sourcefile = pyxfile[:-3] + extension
                    msg = (
                        f"{extension}-source file '{sourcefile}' not found.\n"
                        "Run 'setup.py cython' before sdist."
                    )
                    assert os.path.isfile(sourcefile), msg
        sdist_class.run(self)

> 

I then tried to run this:  python setup.py cython. -- still error 

> 

(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % python setup.py cython
  File "setup.py", line 249
    f"{extension}-source file '{sourcefile}' not found.\n"
                                                         ^
SyntaxError: invalid syntax

> 

Finally, I searched the whole folder and can't find anything called pyxfiles.  

At a total loss here and not sure what to do. I am using a mac with M1 chip. not sure whether would explain anything or not.

Reply:
> I tried in vscode to build the c extension using both base and panda-env

No need to use `base`, just `pandas-dev`. Can you show your complete error message please, and also show the output of `git status` and `git diff origin/shiny-new-feature` to check you haven't accidentally changed something locally?

Reply:
Here is the whole error message: 

(base) jiezheng@Jies-MacBook-Air pandas-jie % conda activate pandas-dev                                
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % git status
On branch shiny-new-feature
Your branch is up to date with 'origin/shiny-new-feature'.

nothing to commit, working tree clean
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % git diff origin/shiny-new-feature
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % python setup.py build_ext -i -j 4
  File "setup.py", line 249
    f"{extension}-source file '{sourcefile}' not found.\n"
                                                         ^
SyntaxError: invalid syntax
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % pytest pandas/tests/generic/test_generic.py 
ImportError while loading conftest '/Users/jiezheng/pandas-jie/pandas/conftest.py'.
pandas/__init__.py:22: in <module>
    from pandas.compat import (
pandas/compat/__init__.py:15: in <module>
    from pandas.compat.numpy import (
pandas/compat/numpy/__init__.py:7: in <module>
    from pandas.util.version import Version
pandas/util/__init__.py:1: in <module>
    from pandas.util._decorators import (  # noqa
pandas/util/_decorators.py:14: in <module>
    from pandas._libs.properties import cache_readonly  # noqa
pandas/_libs/__init__.py:13: in <module>
    from pandas._libs.interval import Interval
E   ModuleNotFoundError: No module named 'pandas._libs.interval'
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie %

Reply:
Thanks @Jiezheng2018 !

Just to confirm, have you already tried:
```
python setup.py build_ext --inplace --force
```
and
```
xcode-select --install
```
?

If so, @WillAyd sorry to tag you again, but I don't know what to suggest here

Reply:
'(base) jiezheng@Jies-MacBook-Air pandas-jie % conda activate pandas-dev                                               
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % python setup.py build_ext --inplace --force

  File "setup.py", line 249
    f"{extension}-source file '{sourcefile}' not found.\n"
                                                         ^
SyntaxError: invalid syntax
(pandas-dev) jiezheng@Jies-MacBook-Air pandas-jie % xcode-select --install

xcode-select: error: command line tools are already installed, use "Software Update" to install updates
'
The above is what I get when running them in VS Code, but on the terminal, it will rebuild setup.py successfully. 
Maybe I need to reinstall my VS Code?

Reply:
I don't know, but if it's working from the terminal let's just use that for now